### PR TITLE
Add tests for sunpy.physics.differential_rotate

### LINKED ISSUE #3239: Need test for time input in differential rotation function
Currently, the `sunpy.physics.differential_rotate` function is only tested for an input observer coordinate. There should also be a set of tests for a new obstime or time interval. This will avoid issues like #3189

### PR DESCRIPTION
**Pull request by dipanshu231099**
### Description

Previously the tests for differential rotation were only considering a new observer. 
I have created tests for differential rotations as the time parameter is altered.

Fixes #3239

### CODE REVIEW

File: sunpy/physics/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (nabobalis) [inline sunpy/physics/.vscode/settings.json:1]: You can remove this file.

File: changelog/4333.trivial.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Added the required tests for `sunpy.physics.differential_rotation.differential_rotate` based on time parameter as input.
```
Comment (nabobalis) [inline changelog/4333.trivial.rst:1]: You can remove this file.

**Comment by dipanshu231099** [comment]
@nabobalis is it mergeable now ?

**Comment by nabobalis** [comment]
This needs two reviews from people who have worked on `sunpy.physics.differential_rotate` before it can be merged.

**Comment by wtbarnes** [comment]
So I know that this persisted prior to this PR, but it would be nice to break these tests up a bit. Rather than having one monolithic `test_differential_rotate` function, I'd prefer that we have a set of tests for this functionality.

As for the bits that test the time input, I think this could probably be parametrized, e.g. for a single function, the two possible input sets for observer and time could be `((new_observer, None), (None, new_time))`.

**Comment by dipanshu231099** [comment]
> So I know that this persisted prior to this PR, but it would be nice to break these tests up a bit. Rather than having one monolithic `test_differential_rotate` function, I'd prefer that we have a set of tests for this functionality.
> 
> As for the bits that test the time input, I think this could probably be parametrized, e.g. for a single function, the two possible input sets for observer and time could be `((new_observer, None), (None, new_time))`.

shall i make a different test definition for time tests then ? @wtbarnes

**Comment by nabobalis** [comment]
Yes.

**Comment by wtbarnes** [comment]
> > So I know that this persisted prior to this PR, but it would be nice to break these tests up a bit. Rather than having one monolithic `test_differential_rotate` function, I'd prefer that we have a set of tests for this functionality.
> > As for the bits that test the time input, I think this could probably be parametrized, e.g. for a single function, the two possible input sets for observer and time could be `((new_observer, None), (None, new_time))`.
> 
> shall i make a different test definition for time tests then ? @wtbarnes

Yes that'd be great! If you're interested in taking a stab at seeing how the existing test code could be broken up, that could make things a bit nicer as well, e.g. having separate tests for all on disk, partially on disk, all of disk.

**Comment by dipanshu231099** [comment]
@wtbarnes i have segregated the different test blocks pre existing ones as well as new ones for the differential_rotate function

**Comment by dipanshu231099** [comment]
@wtbarnes any other thing required in it?

File: sunpy/physics/tests/test_differential_rotation.py [hunk 3]
```diff
@@ -207,13 +218,56 @@ def test_differential_rotate(aia171_test_map, all_off_disk_map, all_on_disk_map,
...
+    # Ignore some invalid NaN comparisions within astropy
+    # (fixed in astropy 4.0.1 https://github.com/astropy/astropy/pull/9843)
+    with np.errstate(invalid='ignore'):
+        dmap = differential_rotate(straddles_limb_map, time=new_time)
+    assert dmap.data.shape[1] < straddles_limb_map.data.shape[1]
+    # The output map should have the same time as the new time now.
+    assert dmap.date == new_time
+
+
+def test_differential_rotate_time_off_disk(all_off_disk_map):
```
Comment (nabobalis) [inline sunpy/physics/tests/test_differential_rotation.py:264]: Is this not the same as `test_differential_rotate_observer_all_off_disk`?

File: sunpy/physics/tests/test_differential_rotation.py [hunk 4]
```diff
@@ -207,13 +218,56 @@ def test_differential_rotate(aia171_test_map, all_off_disk_map, all_on_disk_map,
...
+    # Test a map that is entirely off the disk of the Sun
+    # Should report an error
+    with pytest.raises(ValueError):
+        differential_rotate(all_off_disk_map)
```
Comment (nabobalis) [inline sunpy/physics/tests/test_differential_rotation.py:268]:
```suggestion
        differential_rotate(all_off_disk_map, time=new_time)
```

**Comment by nabobalis** [comment]
Thanks @dipanshu231099

**Pull request merged**